distributed/worker_state_machine.py

```diff
@@ -2670,92 +2942,80 @@ def validate_task(self, ts):
                 self._validate_task_released(ts)
         except Exception as e:
             logger.exception(e)
-            if LOG_PDB:
-                import pdb
-
-                pdb.set_trace()
-
             raise InvalidTaskState(
                 key=ts.key, state=ts.state, story=self.story(ts)
             ) from e
 
-    def validate_state(self):
-        try:
-            assert self.executing_count >= 0
-            waiting_for_data_count = 0
-            for ts in self.tasks.values():
-                assert ts.state is not None
-                # check that worker has task
-                for worker in ts.who_has:
-                    assert worker != self.address
-                    assert ts.key in self.has_what[worker]
-                # check that deps have a set state and that dependency<->dependent links
-                # are there
-                for dep in ts.dependencies:
-                    # self.tasks was just a dict of tasks
-                    # and this check was originally that the key was in `task_state`
-                    # so we may have popped the key out of `self.tasks` but the
-                    # dependency can still be in `memory` before GC grabs it...?
-                    # Might need better bookkeeping
-                    assert dep.state is not None
-                    assert ts in dep.dependents, ts
-                if ts.waiting_for_data:
-                    waiting_for_data_count += 1
-                for ts_wait in ts.waiting_for_data:
-                    assert ts_wait.key in self.tasks
-                    assert (
-                        ts_wait.state
-                        in READY | {"executing", "flight", "fetch", "missing"}
-                        or ts_wait in self._missing_dep_flight
-                        or ts_wait.who_has.issubset(self.in_flight_workers)
-                    ), (ts, ts_wait, self.story(ts), self.story(ts_wait))
-            # FIXME https://github.com/dask/distributed/issues/6319
-            # assert self.waiting_for_data_count == waiting_for_data_count
-            for worker, keys in self.has_what.items():
+    def validate_state(self) -> None:
+        assert len(self.executing) >= 0
```

Comment:
Changes in this method are just a de-indentation.
Error control has remained in `Worker.validate_state`.